pkg/hostagent/dns/dns.go

```diff
@@ -140,26 +140,60 @@ func (h *Handler) handleQuery(w dns.ResponseWriter, req *dns.Msg) {
 				qtype = dns.TypeA
 			}
 			fallthrough
-		case dns.TypeCNAME, dns.TypeA:
+		case dns.TypeA:
+			var err error
+			var addrs []net.IP
+			if _, ok := h.hostToIP[q.Name]; ok {
+				addrs = []net.IP{h.hostToIP[q.Name]}
+			} else {
+				addrs, err = net.LookupIP(q.Name)
```

Comment:
This code will do a CNAME lookup using upstream DNS, but will not resolve names from `h.cnameToHost` here, like the old code did. Why did you change this?

Maybe it doesn't matter, as an `A` record lookup normally doesn't look up `CNAME` records either, but we do this here anyways because there is no `net.LookupA` but only `net.LookupIP`, and handling internal and external CNAMEs differently feels wrong to me.